### PR TITLE
Add XML support for cloudformation commands that lacked it

### LINKED ISSUE #444: boto3: cloudformation JSON response
Same problem like #402 and #427 .

```python
cloudformation = boto3.client('cloudformation')
        stack = cloudformation.create_stack(StackName=name,
                                            TemplateBody='{"AWSTemplateFormatVersion":"2010-09-09","Resources":{"S3LogBucket":{"Type":"AWS::S3::Bucket"}}}')
```

Error: 
```
ResponseParserError: Unable to parse response (not well-formed (invalid token): line 1, column 0), invalid XML received:
{"CreateStackResponse": {"CreateStackResult": {"StackId": "arn:aws:cloudformation:us-east-1:123456789:stack/myFunnyStack/c99845b1-dbc8-4b98-9167-230817390c4a"}}}
```

### PR DESCRIPTION
**Pull request by 2rs2ts**
This lets boto3's cloudformation API work with moto.

A lot of this work was inspired by https://github.com/spulec/moto/pull/404.

fixes #444 

Signed-off-by: Scott Greene <scott.greene@getbraintree.com>

### CODE REVIEW

**Comment by 2rs2ts** [comment]
The same CI failures happen on my machine on master. I don't think the failures are related to my submission.

File: moto/sns/responses.py [hunk 1]
```diff
@@ -12,10 +12,6 @@ class SNSResponse(BaseResponse):
     def backend(self):
         return sns_backends[self.region]
 
-    @property
-    def request_json(self):
-        return 'JSON' in self.querystring.get('ContentType', [])
```
Comment (spulec) [inline moto/sns/responses.py]: Was this intentionally removed?

File: requirements-dev.txt [hunk 2]
```diff
@@ -5,6 +5,8 @@ sure>=1.2.24
 coverage
 freezegun
 flask
-boto3>=1.2.3
-botocore>=1.3.26
-six
\ No newline at end of file
+# botocore 1.3.29 breaks s3 in tests (lambda and s3 tests)
+# so we need to pin a boto3 and botocore revision pair that we know works
```
Comment (spulec) [inline requirements-dev.txt]: I'm confused. This is an issue with botocore 1.3.29 or an issue with moto not working with that version?

File: tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        }
+    ]
+    cf = boto3.resource('cloudformation', region_name='us-east-1')
+    stack = cf.create_stack(
+        StackName="test_stack",
+        TemplateBody=dummy_template_json,
+        Tags=tags,
+    )
+
+    set(stack.tags).should.equal(set(tags))
```
Comment (spulec) [inline tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py]: This line seems to be causing Travis to fail.

**Comment by spulec** [comment]
Thanks for opening. I'm excited for us to get this merged.

File: requirements-dev.txt (hunk 2 above)
Comment (spulec) [inline requirements-dev.txt]: Never mind, just saw the other PR and issue.

File: moto/sns/responses.py (hunk 1 above)
Comment (2rs2ts) [inline moto/sns/responses.py]: Yes, I [added it to the base](https://github.com/spulec/moto/pull/550/files#diff-427c67d098e415aae0a327e597f8c7c7R258).

File: tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py (hunk 3 above)
Comment (2rs2ts) [inline tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py]: Thanks, forgot dicts aren't hashable :sweat_smile:

**Comment by 2rs2ts** [comment]
@spulec Got the build green!

**Comment by spulec** [comment]
Great, thanks!

File: moto/sns/responses.py (hunk 1 above)
Comment (spulec) [inline moto/sns/responses.py]: Awesome

**Pull request merged**